### PR TITLE
PLAT-126333: Added UI tests for TimePicker

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
Enact-DCO-1.0-Signed-off-by: Daniel Stoian daniel.stoian@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
 - Added ui tests for TimePicker

### Links
[//]: # (Related issues, references)
PLAT-126333

### Comments

### CODE REVIEW

File: tests/ui/specs/TimePicker/Agate-TimePicker-Page.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	active (picker) {
+		return element('.internal_Picker_Picker_active', picker);
+	}
+}
+
+class AgateTimePickerPage extends Page {
+	constructor () {
+		super();
+		this.title = 'TimePicker Test';
+		this.components = {};
```
Comment (adrian-cocoara-lgp) [inline tests/ui/specs/TimePicker/Agate-TimePicker-Page.js:42]: If you refer the Picker instances in `specs.js` files as `this.components.timePickerDefault`, `this.components.timePickerWithDefaultValue` etc., I don't think this is needed.

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

File: tests/ui/specs/TimePicker/TimePickerPage.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		this.title = 'TimePicker Test';
+		this.components = {};
+		this.components.timePickerDefault = new PickerInterface('timePickerDefault');
+		this.components.timePickerWithDefaultValue = new PickerInterface('timePickerWithDefaultValue');
+		this.components.timePickerDisabled = new PickerInterface('timePickerDisabled');
+		this.components.timePickerDisabledWithDefaultValue = new PickerInterface('timePickerDisabledWithDefaultValue');
+	}
+
+	open (urlExtra) {
+		super.open('Agate-TimePicker-View', urlExtra);
```
Comment (jeonghee27) [inline tests/ui/specs/TimePicker/TimePickerPage.js:50]: After changing the file names, it was not changed.

File: tests/ui/specs/TimePicker/TimePicker-specs.js [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+			it('should have hour-minute-meridiem order', function () {
+				browser.waitUntil(() => timePicker.decrementer(timePicker.hour).isFocused(), {timeout: 1500,  timeoutMsg: 'hour focused'});
+				Page.spotlightRight();
+				browser.waitUntil(() => timePicker.decrementer(timePicker.minute).isFocused(), {timeout: 1500,  timeoutMsg: 'minute focused'});
+				Page.spotlightRight();
+				browser.waitUntil(() => timePicker.decrementer(timePicker.meridiem).isFocused(), {timeout: 1500,  timeoutMsg: 'meridiem focused'});
+			});
+
+			describe('5-way', function () {
+				it('should decrease the hour when decrementing the picker', function () {
```
Comment (jeonghee27) [inline tests/ui/specs/TimePicker/TimePicker-specs.js:27]: I got test fail on my local testing. 
Could you please test again on you side?  @daniel-stoian-lgp @adrian-cocoara-lgp

Comment (daniel-stoian-lgp) [reply tests/ui/specs/TimePicker/TimePicker-specs.js:27]: I pushed a fix for this bug

**Review by jeonghee27 — APPROVED** [review]
LGTM. sorry for late review

File: tests/ui/specs/TimePicker/TimePicker-specs.js [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+	describe('LTR locale', function () {
+		beforeEach(function () {
+			Page.open();
+		});
+
+		describe('default', function () {
+			const timePicker = Page.components.timePickerDefault;
+
+			it('should have correct title', function () {
```
Comment (MikyungKim) [inline tests/ui/specs/TimePicker/TimePicker-specs.js:14]: I'm not sure why we need this..

Comment (daniel-stoian-lgp) [reply tests/ui/specs/TimePicker/TimePicker-specs.js]: right. i'll removed it

File: tests/ui/specs/TimePicker/TimePicker-specs.js [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+				});
+
+				it('should decrease the minute when decrementing the picker', function () {
+					const {minute} = extractValues(timePicker);
+					timePicker.decrementer(timePicker.minute).click();
+					browser.waitUntil(() => timePicker.decrementer(timePicker.minute).isFocused(), {timeout: 1500,  interval: 100});
+					const {minute: value} = extractValues(timePicker);
+					const expected = minute !== 0 ? minute - 1 : 59;
+					expect(value).to.equal(expected);
+				});
```
Comment (MikyungKim) [inline tests/ui/specs/TimePicker/TimePicker-specs.js:127]: Why don't you add the below test?
```
it('should change the meridiem on hour boundaries', function () {
	const value = timePicker.timeLabel;
	// 12 hours ought to change the value text if meridiem changes
	for (let i = 12; i; i -= 1) {
		timePicker.decrementer(timePicker.hour).click();
	}
	expect(value !== timePicker.timeLabel).to.be.true();
});
```

Comment (daniel-stoian-lgp) [reply tests/ui/specs/TimePicker/TimePicker-specs.js:127]: we don't have timeLabel in Agate. 
But, we could verifiy that the Meridiem label has changed

Comment (MikyungKim) [reply tests/ui/specs/TimePicker/TimePicker-specs.js:127]: Ah, I see. yes. we could.

Comment (daniel-stoian-lgp) [reply tests/ui/specs/TimePicker/TimePicker-specs.js:144]: I added tests for this matter for 5-way and pointer

File: tests/ui/specs/TimePicker/TimePicker-specs.js [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+
+		it('should increment hours from 0 to 23', function () {
+			for (let i = 0; i < 24; i++) {
+				timePicker.incrementer(timePicker.hour).click();
+			}
+			const {hour} = extractValues(timePicker);
+			expect(hour).to.equal(23);
+		});
+
+		it('should decrement hours from 23 to 0', function () {
```
Comment (MikyungKim) [inline tests/ui/specs/TimePicker/TimePicker-specs.js:274]: I think these tests aren't meaningful here.
Let's add "increment hours from 23 to 0" and "decrement hours from 0 to 23" from PLAT-124934 with 'wrap' prop.

Comment (MikyungKim) [reply tests/ui/specs/TimePicker/TimePicker-specs.js:274]: Or we could just merge this and add additional ones from there.

Comment (daniel-stoian-lgp) [reply tests/ui/specs/TimePicker/TimePicker-specs.js:274]: please merge this one and I'll update 124934 right after

Comment (MikyungKim) [reply tests/ui/specs/TimePicker/TimePicker-specs.js:274]: 👍

**Pull request merged**